tests/backup/backup_test.go

```diff
@@ -2722,6 +2722,292 @@ var _ = Describe("{BackupAlternatingBetweenLockedAndUnlockedBuckets}", func() {
+			log.InfoD("internalPolicyStatus - [%v]", intervalPolicyStatus)
+			dash.VerifyFatal(intervalPolicyStatus, nil, "Creating interval schedule policy")
+
+			//prerule
+			preRuleStatus, preRuleName, err := Inst().Backup.CreateRuleForBackup(bkpNamespaces[0], orgID, "pre")
+			log.FailOnError(err, "Creating pre rule for deployed apps failed")
+			dash.VerifyFatal(preRuleStatus, true, "Verifying pre rule for backup")
+
+			//postrule
+			postRuleStatus, postRuleName, err := Inst().Backup.CreateRuleForBackup(bkpNamespaces[0], orgID, "post")
+			log.FailOnError(err, "Creating post rule for deployed apps failed")
+			dash.VerifyFatal(postRuleStatus, true, "Verifying Post rule for backup")
+
+			preRuleUid, err := Inst().Backup.GetRuleUid(orgID, ctx, preRuleName)
+			log.FailOnError(err, "Generating pre rule UID for deployed apps failed")
+			postRuleUid, err := Inst().Backup.GetRuleUid(orgID, ctx, postRuleName)
+			log.FailOnError(err, "Generating post rule UID for deployed apps failed")
+
+			log.InfoD("Fetching Schedule Backup uid")
+			schPolicyUid, err = Inst().Backup.GetSchedulePolicyUid(orgID, ctx, periodicPolicyName)
+			log.FailOnError(err, "Fetching schedule policy failed")
+
+			//CreateSchedule backup
+			log.InfoD("backup schedule name - %v", backupName)
+			err = CreateScheduleBackup(backupName, SourceClusterName, backupLocationName, backupLocationUID, []string{bkpNamespaces[0]}, nil, orgID, preRuleName, preRuleUid, postRuleName, postRuleUid, periodicPolicyName, schPolicyUid, ctx)
+			log.FailOnError(err, "Generating Schedule Backup")
+		})
+
+		Step("Validate the Access toggle ", func() {
+			accesses = append(accesses, ViewOnlyAccess, RestoreAccess, FullAccess)
+			log.InfoD("Accesses - %v", accesses)
+			// Get user context
+			ctxNonAdmin, err := backup.GetNonAdminCtx(userName, password)
+			log.FailOnError(err, "Fetching %s ctx", userName)
+
+			// Register Source and Destination cluster
+			log.InfoD("Registering Source and Destination clusters from user context")
+			err = CreateSourceAndDestClusters(orgID, "", "", ctxNonAdmin)
+			dash.VerifyFatal(err, nil, "Creating source and destination cluster")
+
+			for _, accessLevel := range accesses {
+
+				log.InfoD("Current Access is  %v", accessLevel)
+
+				log.InfoD("share all backups with %v in source cluster with a user", accessLevel)
+				err := ClusterUpdateBackupShare(SourceClusterName, nil, []string{userName}, accessLevel, true, ctx)
+				log.FailOnError(err, "Failed sharing all backups for cluster [%s]", SourceClusterName)
+
+				userBackups, err := GetAllBackupsForUser(userName, password)
+				log.InfoD("len(userBackups %v", len(userBackups))
+				log.FailOnError(err, "Fail on Fetching backups for %s", userName)
+
+				restoreName := fmt.Sprintf("%s-%v", RestoreNamePrefix, time.Now().Unix())
+				ValidateSharedBackupWithUsers(userName, ctxNonAdmin, accessLevel, userBackups[len(userBackups)-1], restoreName)
```

Comment:
ValidateSharedBackupWithUsers should be in the end . After checking if its full access, waiting for 15 mins to get the new backup ....